src/field_spec.h

```diff
@@ -92,6 +92,8 @@ typedef struct FieldSpec {
 
   // Vector similarity index parameters.
   VecSimParams vecSimParams;
+  // expected size of vector blob.
+  size_t expBlobSize;
```

Comment:
Why adding a new field to the generic struct? `expBlobSize` can be deduced from `vecSimParams`

Reply:
yes, but do we want to calculate it on every insertion?

Reply:
Good point.
We can add it in `VecSimParams` as `int` after `algo` and with padding, not consume more memory. WDYT?

Reply:
I see. I would adding `expBlobSize` to `vecSimParams` - but that will require changes in VecSim -  so maybe we should leave it for now...

Reply:
I think it’s not a parameter for creating the index so it doesn’t need to be there, but maybe we can add to the FieldSpec a union? There are few things that relate only for specific indexing types

Reply:
this is not related to index creating so it should not be in `vecSimParams`. Same as tag field has 'tagSep', I think for now it is a valid solution

Reply:
> I think it’s not a parameter for creating the index so it doesn’t need to be there, but maybe we can add to the FieldSpec a union? There are few things that relate only for specific indexing types like the tag only params

Agree, union it's a good idea for future refactor